File: design-patterns/program/src/7-Object-Pool/DatabasePool.php
<?php
namespace ObjectPool;

use ObjectPool\Connection;

class DatabasePool {
    /**
     * @var Pool
     */
    private static $pool = array();

    public function __construct() {
        for ($i = 0; $i < 5; $i++) { 
            $connection = new Connection("localhost", "root", "root");
            array_push(self::$pool, $connection);
        }
    }

    public static function getConnection(): Connection {
        try {
            if (count(self::$pool) === 0) {
                throw new \Exception("Pool connection is Empty!");
            } else {
                return array_shift(self::$pool);
            }
        } catch (\Exception $e) {
            echo $e->getMessage();
        }
    }

    public static function close(Connection $connection) {
        array_push(self::$pool, $connection);
    }
}

File: design-patterns/program/src/7-Object-Pool/java/DatabasePool.java
import java.util.ArrayList;
import java.util.List;

private class DatabasePool {
    private static List<Connection> pool = new ArrayList<>();

    static {
        for (int i = 0; i < 100; i++) {
            Connection connection = new Connection("localhost", "root", "root");
            pool.add(connection);
        }
    }

    public static Connection getConnection() {
        if (pool.isEmpty()) {
            throw new RuntimeException("Connection is empty");
        } else {
            Connection connection = pool.remove(0);
            return connection;
        }
    }

    public static void close(Connection connection) {
        pool.add(connection);
    }
}